build.rs

```diff
@@ -547,10 +548,24 @@ fn obj_path(out_dir: &Path, src: &Path) -> PathBuf {
 
 fn configure_cc(c: &mut cc::Build, target: &Target, c_root_dir: &Path, include_dir: &Path) {
     let compiler = c.get_compiler();
-    // FIXME: On Windows AArch64 we currently must use Clang to compile C code
-    let compiler = if target.os == WINDOWS && target.arch == AARCH64 && !compiler.is_like_clang() {
-        let _ = c.compiler("clang");
-        c.get_compiler()
+    // FIXME: On Windows AArch64, ring C code cannot be compiled using cl.exe, but must be compiled
+    // using the LLVM toolchain.
+    let compiler = if target.os == WINDOWS && target.arch == AARCH64 {
+        match compiler.family() {
+            // If the user is already using clang or clang-cl, nothing needs to be replaced
+            cc::ToolFamily::Clang { .. } | cc::ToolFamily::Msvc { clang_cl: true, .. } => compiler,
+            // If cl.exe is used, replace that with clang-cl which is most likely to be compatible
+            // with flags that are already in place (i.e. custom CFLAGS that the user provided to cc).
+            cc::ToolFamily::Msvc {
+                clang_cl: false, ..
+            } => {
+                // FIXME: This requires clang-cl to be available in PATH, regardless of any explicit
+                // or custom path that the user might have provided to cc via the CC flag.
+                c.compiler("clang-cl");
+                c.get_compiler()
+            }
+            t => panic!("Unsupported tool `{t:?}` for ARM64 Windows compilation"),
```

Comment:
Similarly, I think we just keep going with whatever the compiler is, and see what happens.